dashboards_bundlers/server_upload/__init__.py

```diff
@@ -18,6 +22,35 @@ def bundle(handler, abs_nb_path):
     notebook_basename = os.path.basename(abs_nb_path)
     notebook_name = os.path.splitext(notebook_basename)[0]
 
+    # Python 2/3 compatible try/finally to cleanup a temp working directory
+    tmp_dir = tempfile.mkdtemp()
+    try:
+        output_dir = os.path.join(tmp_dir, notebook_name)
+        bundled = make_upload_bundle(abs_nb_path, output_dir, handler.tools)
+        send_file(bundled, notebook_name, handler)
+    finally:
+        shutil.rmtree(tmp_dir, True)
+
+def make_upload_bundle(abs_nb_path, staging_dir, tools):
+    # Clean up bundle dir if it exists
+    shutil.rmtree(staging_dir, True)
+    os.makedirs(staging_dir)
+
+    # Include the notebook as index.ipynb to make the final URL cleaner
+    # and for consistency
+    shutil.copy2(abs_nb_path, os.path.join(staging_dir, 'index.ipynb'))
+    # Include frontend files referenced via the jupyter_cms bundle mechanism
+    bundle_file_references(staging_dir, abs_nb_path, tools)
+    bundle_declarative_widgets(staging_dir, abs_nb_path)
```

Comment:
The behavior within this function is busted slightly. The line https://github.com/jupyter-incubator/dashboards_bundlers/blob/master/dashboards_bundlers/local_deploy/__init__.py#L160 checks if there is at least `urth-core-import`. But, as I found out the hard way in #21 with the simple widget notebook that only uses `urth-core-bind`, you can use the decl widgets without any imports. 

I suggest switching the logic to treat any `urth-core-` references as an indicator that the widgets need to be bundled.

/cc @Lull3rSkat3r